metrics/metrics.go

```diff
@@ -24,6 +24,9 @@ import (
 
 // The Provider interface defines an interface for measuring metrics.
 type Provider interface {
+	// RecordPartialChecksum records the checksum of a partial template's output.
+	RecordPartialChecksum(name, checksum string)
```

Comment:
This all looks good, but I'm not a big fan of the method name here. Not very important, but I would have prefered `AddTemplateChecksum` -- we may use it for other templates in the future.

Reply:
Done.